### PR TITLE
fix some bug

### PR DESCRIPTION
**Pull request by min-zh**
1. support more parameters of neonsan create_volume
2. judge if device readable, before mount
3. keep idempotent of node operation
4. add correct parameters for clone
5. fix documents error
6. add install script of neonsan-plugin

### CODE REVIEW

File: pkg/service/node.go [hunk 1]
```diff
@@ -52,15 +52,24 @@ func (s *service) NodeStageVolume(ctx context.Context, req *csi.NodeStageVolumeR
...
-	if err != nil {
-		return nil, status.Error(codes.Internal, err.Error())
-	}
+
+	// for idempotent, if device not empty, volume has already attached
 	devicePath, err := s.storageProvider.NodeGetDevice(volumeID)
 	if err != nil {
 		return nil, status.Error(codes.Internal, err.Error())
 	}
+	if len(devicePath) == 0{
```
Comment (zheng1) [inline pkg/service/node.go:61]:
```suggestion
	if len(devicePath) == 0 {
```

File: pkg/service/node.go [hunk 2]
```diff
@@ -109,6 +118,16 @@ func (s *service) NodeUnstageVolume(ctx context.Context, req *csi.NodeUnstageVol
 		klog.Errorf("Volume %s still mounted in instance %s", volumeID, s.option.NodeId)
 		return nil, status.Error(codes.Internal, "unmount failed")
 	}
+
+	devicePath, err := s.storageProvider.NodeGetDevice(volumeID)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	// for idempotent, if device is empty, the volume has already detached
+	if len(devicePath) == 0{
```
Comment (zheng1) [inline pkg/service/node.go:127]:
```suggestion
	if len(devicePath) == 0 {
```

File: pkg/storage/neonsan/api/api.go [hunk 3]
```diff
@@ -208,20 +236,25 @@ func ListSnapshot(confFile, poolName, volumeName, snapshotName string) (*Snapsho
...
 		switch val := v.Field(k).Interface().(type) {
 		case int:
-			sb.WriteString(strconv.Itoa(val))
+			parameter[t.Field(k).Tag.Get(`json`)] = strconv.Itoa(val)
 		case int64:
-			sb.WriteString(strconv.Itoa(int(val)))
+			parameter[t.Field(k).Tag.Get(`json`)] = strconv.Itoa(int(val))
 		case string:
-			sb.WriteString(val)
+			parameter[t.Field(k).Tag.Get(`json`)] = val
```
Comment (zheng1) [inline pkg/storage/neonsan/api/api.go:247]:
```suggestion
			parameter[t.Field(k).Tag.Get("json")] = val
```

File: pkg/storage/neonsan/api/api.go [hunk 4]
```diff
@@ -44,12 +44,28 @@ type Volume struct {
...
+type VolumeForClone struct {
+	ID                  int    `json:"id"`
+	Size                int    `json:"size"`
+	ReplicationCount    int    `json:"replication_count"`
+	MinReplicationCount int    `json:"min_replication_count"`
+	Role                string `json:"role"`
+	MaxBs               int    `json:"max_bs"`
+	Encrypte            string `json:"encrypte"`
+	KeyName             string `json:"key_name"`
+	Rg                  string `json:"rg"`
```
Comment (zheng1) [inline pkg/storage/neonsan/api/api.go:68]: What is RG meaning? Should it change to full name?

Comment (min-zh) [reply pkg/storage/neonsan/api/api.go:68]: I keep the parameter as same as the api document from of neonssan-api.

File: pkg/service/node.go (hunk 2 above)
Comment (min-zh) [reply pkg/service/node.go]: ok, i'll fix it

File: pkg/storage/neonsan/api/api.go (hunk 3 above)
Comment (min-zh) [reply pkg/storage/neonsan/api/api.go]: ok, i'll fix it

**Pull request merged**